fe/fe-core/src/main/java/com/starrocks/connector/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -0,0 +1,111 @@
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import com.starrocks.server.AbstractTableFactory;
+import com.starrocks.server.TableFactoryProvider;
+import com.starrocks.sql.ast.CreateTableStmt;
+import com.starrocks.sql.common.EngineType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// TODO add meta cache
+public class ElasticsearchMetadata
+        implements ConnectorMetadata {
+    private static final Logger LOG = LogManager.getLogger(EsTable.class);
+
+
+    private final EsRestClient esRestClient;
+    private final Map<String, String> properties;
+    private final String catalogName;
+    public static final String DEFAULT_DB = "default";
+    public static final long DEFAULT_DB_ID = 1L;
+
+    public ElasticsearchMetadata(EsRestClient esRestClient, String catalogName, Map<String, String> properties) {
+        this.esRestClient = esRestClient;
+        this.catalogName = catalogName;
+        this.properties = properties;
+    }
+
+    @Override
+    public List<String> listDbNames() {
+        return Arrays.asList(DEFAULT_DB);
+    }
+
+    @Override
+    public List<String> listTableNames(String dbName) {
+        return esRestClient.listTables();
+    }
+
+    @Override
+    public Database getDb(String dbName) {
+        return new Database(DEFAULT_DB_ID, DEFAULT_DB);
+    }
+
+    @Override
+    public Table getTable(String dbName, String tblName) {
+        if (!DEFAULT_DB.equalsIgnoreCase(dbName)) {
+            return null;
+        }
+        return toEsTable(esRestClient, properties, catalogName, tblName);
+    }
+
+    public static EsTable toEsTable(EsRestClient esRestClient,
+                                    Map<String, String> properties,
+                                    String catalogName,
+                                    String tableName) {
+        try {
+            List<ColumnDef> columns = EsUtil.convertColumnSchema(esRestClient, tableName);
+            properties.put(EsTable.INDEX, tableName);
+            Database db = new Database(DEFAULT_DB_ID, DEFAULT_DB);
+            TableName table = new TableName(catalogName, DEFAULT_DB,  tableName);
+            db.setCatalogName(catalogName);
+            CreateTableStmt tableStmt = new CreateTableStmt(false,
+                    false,
+                    table,
+                    columns,
+                    EngineType.ELASTICSEARCH.name(),
+                    new KeysDesc(),
+                    null,
+                    null,
+                    properties,
+                    new HashMap<>(),
+                    "created by internal es catalog");
+
+            AbstractTableFactory tableFactory = TableFactoryProvider.getFactory(EngineType.ELASTICSEARCH.name());
+            EsTable esTable = (EsTable) tableFactory.createTable(null, db, tableStmt);
```

Comment:
you can't use this interface in the catalog. because it will record edit log when getting unique tableId. If follower performs this operator, it will crash.